### PR TITLE
add infra node ip to allowalltcp rule after deployment

### PR DESCRIPTION
**Pull request by YinYangOfDao**
@Anbang-Hu  need to merge the rule creating commands to one function after all current PR merged, plz remind me after all PRs that contains edition to cloud_init_aztools.py merged

### CODE REVIEW

File: src/ClusterBootstrap/cloud_init_aztools.py [hunk 1]
```diff
@@ -535,6 +535,26 @@ def vm_interconnects(config, args):
                config["cloud_config_nsg_rules"]["nfs_allow_master"]["tcp_port_ranges"],
                allowed_incoming_infra_ips
                )
+
+    restricted_source_address_prefixes = "'*'"
+    if "restricted_source_address_prefixes" in config["cloud_config_nsg_rules"]:
+        restricted_source_address_prefixes = config["cloud_config_nsg_rules"]["restricted_source_address_prefixes"]
+        if isinstance(restricted_source_address_prefixes, list):
+            restricted_source_address_prefixes = " ".join(
+                keep_widest_subnet(infra_ip_list + list(set(restricted_source_address_prefixes))))
```
Comment (Anbang-Hu) [inline src/ClusterBootstrap/cloud_init_aztools.py]: Can you change this in `deploy.py` too?

Comment (YinYangOfDao) [reply src/ClusterBootstrap/cloud_init_aztools.py]: added

File: src/ClusterBootstrap/az_tools.py [hunk 2]
```diff
@@ -704,6 +704,28 @@ def vm_interconnects():
...
 
+    restricted_source_address_prefixes = "'*'"
+    if "restricted_source_address_prefixes" in config["cloud_config_nsg_rules"]:
+        restricted_source_address_prefixes = config["cloud_config_nsg_rules"]["restricted_source_address_prefixes"]
+        if isinstance(restricted_source_address_prefixes, list):
+            restricted_source_address_prefixes = " ".join(
+                utils.keep_widest_subnet(infra_ip_list + list(set(restricted_source_address_prefixes))))
+
+    cmd = """
+        ; az network nsg rule update \
```
Comment (Anbang-Hu) [inline src/ClusterBootstrap/az_tools.py]: Let's try to put `az` commands into `az_utils.py` for reuse in a separate PR.

File: src/ClusterBootstrap/az_tools.py [hunk 3]
```diff
@@ -704,6 +704,28 @@ def vm_interconnects():
     output = utils.exec_cmd_local(cmd)
     print(output)
 
+    restricted_source_address_prefixes = "'*'"
+    if "restricted_source_address_prefixes" in config["cloud_config_nsg_rules"]:
+        restricted_source_address_prefixes = config["cloud_config_nsg_rules"]["restricted_source_address_prefixes"]
+        if isinstance(restricted_source_address_prefixes, list):
+            restricted_source_address_prefixes = " ".join(
+                utils.keep_widest_subnet(infra_ip_list + list(set(restricted_source_address_prefixes))))
```
Comment (Anbang-Hu) [inline src/ClusterBootstrap/az_tools.py]: `infra_ip_list` does not exist. @YinYangOfDao can you help fix?

**Pull request merged**